graphql/util.go

```diff
@@ -519,3 +526,18 @@ func canRestartTask(ctx context.Context, at *restModel.APITask) (*bool, error) {
 	canRestart := !utility.StringSliceContains(nonrestartableStatuses, *at.Status) || at.Aborted || (at.DisplayOnly && *taskBlocked)
 	return &canRestart, nil
 }
+
+func getAllTaskStatuses(tasks []task.Task) []string {
+	statusesMap := map[string]bool{}
+	for _, task := range tasks {
+		statusesMap[task.Status] = true
+	}
+	statusesArr := []string{}
+	for key := range statusesMap {
+		statusesArr = append(statusesArr, key)
+	}
+	sort.SliceStable(statusesArr, func(i, j int) bool {
+		return statusesArr[i] < statusesArr[j]
```

Comment:
do these need to be sorted? I think the status selector component in Spruce maintains the ordering in the dropdown.

Reply:
makes for more predictable responses, especially while testing. also takes a 1/1000 of a millisecond.

Reply:
also, tests kept on failing and it may have been due to order not being guaranteed from an object, which is the original data structure before converting it to an array. sorting alphabetically guaranteed the order; therefore preventing flakey tests.